ehsm_kms_service/secret_manager_apis.js

```diff
@@ -0,0 +1,212 @@
+    base64_decode
+} = require('./function')
+const {
+    cryptographic_apis,
+} = require('./apis')
+const ehsm_napi = require('./ehsm_napi')
+
+// get a string parameter from payload and base64 if you need
+function getParam_String(payload, key, needBase64) {
+    if (needBase64 == undefined) {
+        needBase64 = true;
+    }
+    let val = payload[key]
+    if (val != undefined) {
+        val = String(val)
+        if (needBase64) {
+            val = base64_encode(val)
+        }
+    }
+    return val
+}
+
+//check param is string and required
+function checkStringParam(param, required) {
+    if (param == '' || param == undefined) {
+        if (required) {
+            return false
+        } else {
+            return true
+        }
+    } else {
+        if (typeof (param) == 'string') {
+            return true
+        } else {
+            return false
+        }
+    }
+}
+
+function check_encryptionKeyId_format(encryptionKeyId) {
+    if (!((/^([a-z\d]{8}-[a-z\d]{4}-[a-z\d]{4}-[a-z\d]{4}-[a-z\d]{12})$/).test(encryptionKeyId))) {
+        return false
+    }
+    return true
+}
+
+function check_rotationInterval_format(rotationInterval) {
+    if (!((/^(\d{1,4}[d,h,m,s]{1})$/).test(rotationInterval))) {
+        return false
+    }
+    return true
+}
+
+const queryKeyId = (appid, keyid) => {
+    return {
+        selector: {
+            creator: appid,
+            keyid
+        },
+        fields: ['keyBlob'],
+        limit: 1
+    }
+}
+
+const queryDefaultCMK = (appid) => {
+    return {
+        selector: {
+            appid
+        },
+        fields: ['sm_default_cmk'],
+        limit: 1
+    }
+}
+
+//create Secret
+const createSecret = async (res, appid, payload, DB) => {
+    try {
+        let secretData = getParam_String(payload, 'secretData')
+        let secretName = getParam_String(payload, 'secretName')
+        let encryptionKeyId = getParam_String(payload, 'encryptionKeyId', false)
+        let description = getParam_String(payload, 'description')
+        let rotationInterval = getParam_String(payload, 'rotationInterval', false)
+        const createTime = new Date().getTime()
+        let keyBlob = ''
+        let nextRotationDate = ''
+        if (!checkStringParam(secretData, true)) {
+            res.send(_result(400, 'secretData cannot be empty and must be string'))
+            return
+        }
+        if (!checkStringParam(secretName, true)) {
+            res.send(_result(400, 'secretName cannot be empty and must be string'))
+            return
+        }
+        if (!checkStringParam(description, false)) {
+            res.send(_result(400, 'description must be string'))
+            return
+        }
+        if (!checkStringParam(rotationInterval, false)) {
+            res.send(_result(400, 'rotationInterval must be string'))
+            return
+        }
+        if (!check_encryptionKeyId_format(encryptionKeyId) && encryptionKeyId != '' && encryptionKeyId != undefined) {
```

Comment:
seems the logic is wrong?

is the check required: encryptionKeyId != '', can be covered by regex?  or move to the function check_encryptionKeyId_format() ?